UnityProject/Assets/Scripts/Health/Living/LivingHealthBehaviour.cs

```diff
@@ -393,20 +490,92 @@ private bool NotSuitableForDeath()
+
+	// Stats are separated so that the server only updates the area of concern when needed
+
+	/// <summary>
+	/// Updates the main health stats from the server via NetMsg
+	/// </summary>
+	public void UpdateClientHealthStats(int overallHealth, ConsciousState consciousState)
 	{
-		Debug.Log("TODO PRIORITY: Food should no longer heal, instead it should cure hunger");
-		if (amount <= 0)
+		OverallHealth = overallHealth;
+		ConsciousState = consciousState;
+	//	Logger.Log($"Update stats for {gameObject.name} OverallHealth: {overallHealth} ConsciousState: {consciousState.ToString()}", Category.Health);
+		CheckDeadCritStatus();
+	}
+
+	/// <summary>
+	/// Updates the respiratory health stats from the server via NetMsg
+	/// </summary>
+	public void UpdateClientRespiratoryStats(bool isBreathing, bool isSuffocating)
+	{
+		respiratorySystem.UpdateClientRespiratoryStats(isBreathing, isSuffocating);
+	//	Logger.Log($"Update stats for {gameObject.name} isBreathing: {isBreathing} isSuffocating {isSuffocating}", Category.Health);
+
+		CheckDeadCritStatus();
+	}
+
+	/// <summary>
+	/// Updates the blood health stats from the server via NetMsg
+	/// </summary>
+	public void UpdateClientBloodStats(int heartRate, int bloodVolume, int oxygenLevel, int toxinLevel)
+	{
+		bloodSystem.UpdateClientBloodStats(heartRate, bloodVolume, oxygenLevel, toxinLevel);
+		CheckDeadCritStatus();
+	}
+
+	/// <summary>
+	/// Updates the brain health stats from the server via NetMsg
+	/// </summary>
+	public void UpdateClientBrainStats(bool isHusk, int brainDamage)
+	{
+		if (brainSystem != null)
 		{
-			return;
+			brainSystem.UpdateClientBrainStats(isHusk, brainDamage);
+			CheckDeadCritStatus();
 		}
-		OverallHealth += amount;
+	}
 
-		if (OverallHealth > maxHealth)
+	/// <summary>
+	/// Updates the bodypart health stats from the server via NetMsg
+	/// </summary>
+	public void UpdateClientBodyPartStats(BodyPartType bodyPartType, int bruteDamage, int burnDamage)
+	{
+		var bodyPart = FindBodyPart(bodyPartType);
+		if (bodyPart != null)
 		{
-			OverallHealth = maxHealth;
+		//	Logger.Log($"Update stats for {gameObject.name} body part {bodyPartType.ToString()} BruteDmg: {bruteDamage} BurnDamage: {burnDamage}", Category.Health);
+
+			bodyPart.UpdateClientBodyPartStat(bruteDamage, burnDamage);
+		}
+	}
+
+	/// ---------------------------
+	/// MISC Functions:
+	/// ---------------------------
+
+	///<summary>
+	/// If Harvesting is allowed (for pete the goat for example)
+	/// then spawn the butchered results
+	/// </summary>
+	[Server]
+	public void Harvest()
+	{
+		foreach (GameObject harvestPrefab in butcherResults)
+		{
+			ItemFactory.SpawnItem(harvestPrefab, transform.position, transform.parent);
 		}
+		EffectsFactory.Instance.BloodSplat(transform.position, BloodSplatSize.medium);
+		//Remove the NPC after all has been harvested
+		ObjectBehaviour objectBehaviour = gameObject.GetComponent<ObjectBehaviour>();
+		objectBehaviour.visibleState = false;

```

Comment:
Perhaps it's better to networkdestroy the harvested corpse instead of  making it invisible?

Reply:
Yeah only if we were returning these NPC's back to the pool. Which we probably don't need to do. Unless we have a game mode where we have a herd of petes that never stop coming.

Reply:
Simply making bodies invisible will lead to unwanted behaviour further on. 
At least use PlayerSync/CustomNetTransform.DisappearFromWorldServer() method so that it'd be considered hidden (at InvalidPos)

Reply:
fixed